File: 1183-maximum-number-of-ones/1183-maximum-number-of-ones.py
class Solution:
    def maximumNumberOfOnes(self, width: int, height: int, sideLength: int, maxOnes: int) -> int:
        count = []
        for i in range(sideLength):
            for j in range(sideLength):
                count += [((width - i - 1) // sideLength + 1) * ((height - j - 1) // sideLength + 1)]
        count = sorted(count, reverse = True)
        return sum(count[:maxOnes])

File: 1183-maximum-number-of-ones/1183-maximum-number-of-ones.cpp
class Solution {
public:
    int maximumNumberOfOnes(int width, int height, int sideLength, int maxOnes) {
        priority_queue<int> pq;
        for(int r = 0; r < sideLength; ++r)
            for(int c = 0; c < sideLength; ++c)
                pq.push((1 + (width - 1 - r) / sideLength) * (1 + (height - 1 - c) / sideLength));
                
        int res = 0;
        int i = 0;
        while(!pq.empty() && i < maxOnes) {
            res += pq.top(); pq.pop(); ++i;
        }
        return res;
    }
};